azure-keyvault/examples/test_examples_keyvault.py

```diff
@@ -0,0 +1,197 @@
+
+
+def test_example_secret_crud_operations():
+   
+    vault_client = create_vault_client()
+    client = vault_client.secrets
+
+    try:
+        # [START get_secret]
+
+        # get secret with version
+        client.get_secret('secret name', 'secret_version')
+
+        # [END get_secret]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START set_secret]
+        from dateutil import parser as date_parse
+
+        expires = date_parse.parse('2050-02-02T08:00:00.000Z')
+        client.set_secret('secret name', 'secret_value', enabled=False, expires=expires)
+
+        # [END set_secret]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START update_secret_attributes]
+
+        content_type = 'text/plain'
+        expires = date_parse.parse('2050-02-02T08:00:00.000Z')
+        tags = {'foo': 'updated tag'}
+        
+        secret_bundle = client.update_secret_attributes(
+            'secret name', 'secret_version',
+            content_type=content_type,
+            expires=expires,
+            tags=tags)
+
+        # [END update_secret_attributes]
+    except ClientRequestError:
+        pass
+
+def test_example_secret_list_operations():
+
+    vault_client = create_vault_client()
+    client = vault_client.secrets
+
+    try:
+        # [START list_secrets]
+
+        # get secret with max_page_size
+        client.list_secrets(max_page_size=3)
+
+        # get secret without max_page_size (default = 25)
+        client.list_secrets()
+
+        # [END list_secrets]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START list_secret_versions]
+        import os
+        from azure.keyvault.vault_client import VaultClient
+        
+        # get secret with max_page_size
+        client.list_secret_versions('secret name', max_page_size=26)
+
+        # get secret without max_page_size
+        client.list_secret_versions('secret name')
+
+        # [END list_secret_versions]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START list_deleted_secrets]
+        import os
+        from azure.keyvault.vault_client import VaultClient
+        # TODO: should we put here how to create a vault client first?
+        
+        # get secret with max_page_size
+        client.list_deleted_secrets('secret name', max_page_size=26)
+
+        # get secret without max_page_size
+        client.list_deleted_secrets('secret name')
+
+        # [END list_deleted_secrets]
+    except ClientRequestError:
+        pass
+
+def test_example_secrets_backup_restore():
+
+    vault_client = create_vault_client()
+    client = vault_client.secrets
+
+    try:
+        # [START backup_secret]
+
+        # backup secret
+        secret_backup = client.backup_secret('secret name')
+
+        # [END backup_secret]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START delete_secret]
+
+        # delete secret
+        client.delete_secret('secret name')
```

Comment:
Might be helpful to assign the result to something, to indicate data is returned.